src/main.js

```diff
@@ -102,7 +102,7 @@ var load = loader({
   server: {
     requires: ['cfg', 'router', 'docs'],
     setup: ({cfg, router, docs}) => {
-      let hooksApp = app(cfg.server);
+      let hooksApp = app({docs,cfg.server});
```

Comment:
I don't think this syntax is correct.  Did this pass lint checks for you?

The `app` function expects an object containing options.  The `cfg.server` object fits that description, but now we want to add an extra option to it.  The easiest fix may be to break that out into individual options:

```javascsript
app({
  docs,
  port: cfg.server.port,
  publicUrl: cfg.server.publicUrl,
  /* .. etc. .. */
})
```

However, you can also use the new [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator) if you want to get fancy.

Please make sure the tests pass for you before pushing :)

Reply:
That's OK.  There are some instructions in the README, but it boils down to this: run `yarn` to install the dependencies, then run `yarn test` to run the tests.  That will skip a few since you don't have credentials, but that's OK for now.

Reply:
![screenshot from 2017-12-08 00-34-28](https://user-images.githubusercontent.com/15844831/33733443-cfd7b6a0-dbaf-11e7-81d1-7982b7617178.png)
I get this error when run `yarn`

Reply:
how do I get yarn 0.24.6 . I tried `npm install -g yarn@0.24.6` but that didn't seem to solve the problem

Reply:
Ugh, sorry.  Hooks is using an older version of Node and Yarn.  @alexandrasp is working on a "spring cleaning" patch that will improve those things.

But let's use `npm` until Alex's patch lands:
```
$ npm install
# <.. lots of time installing things ..>
$ npm run test

> taskcluster-hooks@0.0.1 pretest /home/dustin/p/taskcluster-hooks
> npm run compile


> taskcluster-hooks@0.0.1 compile /home/dustin/p/taskcluster-hooks
> babel-compile -p taskcluster src:lib test:.test

Adding preset: babel-preset-taskcluster
Running babel compilation with config:
{
  "presets": [
    "babel-preset-taskcluster"
  ],
  "babelrc": false
}
Deleting existing destintation: lib
Compiling file src/data.js --> lib/data.js
Compiling file src/main.js --> lib/main.js
Compiling file src/nextdate.js --> lib/nextdate.js
Compiling file src/scheduler.js --> lib/scheduler.js
Compiling file src/taskcreator.js --> lib/taskcreator.js
Compiling file src/v1.js --> lib/v1.js
Deleting existing destintation: .test
Compiling file test/api_test.js --> .test/api_test.js
Compiling file test/fake-notify.js --> .test/fake-notify.js
Compiling file test/helper.js --> .test/helper.js
Compiling file test/lint.js --> .test/lint.js
Copying non-js file test/mocha.opts to .test/mocha.opts
Compiling file test/schedule_hooks_test.js --> .test/schedule_hooks_test.js
Compiling file test/scheduler_test.js --> .test/scheduler_test.js
Compiling file test/taskcreator_test.js --> .test/taskcreator_test.js
Compiling file test/test_definition.js --> .test/test_definition.js
Copying non-js file test/validate_test/create-hook-request.json to .test/validate_test/create-hook-request.json
Copying non-js file test/validate_test/schedule-biweekly.json to .test/validate_test/schedule-biweekly.json
Copying non-js file test/validate_test/schedule-daily.json to .test/validate_test/schedule-daily.json
Copying non-js file test/validate_test/schedule-monthly.json to .test/validate_test/schedule-monthly.json
Copying non-js file test/validate_test/schedule-none.json to .test/validate_test/schedule-none.json
Copying non-js file test/validate_test/schedule-weekday.json to .test/validate_test/schedule-weekday.json
Copying non-js file test/validate_test/schedule-weekly.json to .test/validate_test/schedule-weekly.json
Compiling file test/validate_test.js --> .test/validate_test.js

> taskcluster-hooks@0.0.1 test /home/dustin/p/taskcluster-hooks
> mocha .test/lint.js .test/*_test.js



  eslint
    ✓ should have no errors in src/data.js (530ms)
    ✓ should have no errors in src/main.js (145ms)
    ✓ should have no errors in src/nextdate.js
    ✓ should have no errors in src/scheduler.js (155ms)
    ✓ should have no errors in src/taskcreator.js (82ms)
    ✓ should have no errors in src/v1.js (199ms)
    ✓ should have no errors in test/api_test.js (202ms)
    ✓ should have no errors in test/fake-notify.js
    ✓ should have no errors in test/helper.js
    ✓ should have no errors in test/lint.js
    ✓ should have no errors in test/schedule_hooks_test.js
    ✓ should have no errors in test/scheduler_test.js (79ms)
    ✓ should have no errors in test/taskcreator_test.js (57ms)
    ✓ should have no errors in test/test_definition.js
    ✓ should have no errors in test/validate_test.js

  API
    createHook
      ✓ creates a hook (102ms)
      ✓ with invalid scopes
      ✓ succeeds if a matching resource already exists
      ✓ fails if different resource already exists
      ✓ creates associated group
      ✓ without a schedule
      ✓ with a daily schedule
      ✓ fails with invalid schedule
    updateHook
      ✓ updates a hook
      ✓ fails if resource doesn't exist
      ✓ fails if new schedule is invalid
    removeHook
      ✓ removes a hook
      ✓ removed empty groups
    listHookGroups
      ✓ returns valid groups (65ms)
    listHooks
      ✓ lists hooks in the given group only (64ms)
    hook
      ✓ returns a hook
      ✓ fails if no hook exists
    getTriggerToken
      ✓ returns the same token
      ✓ error on requesting token for undefined hook
    getHookSchedule
      ✓ returns {schedule: []} for a non-scheduled task
      ✓ returns the schedule for a -scheduled task
      ✓ fails if no hook exists
    getHookStatus
      ✓ returns "no-fire" for a non-scheduled, non-fired task
      ✓ returns the next date for a scheduled task
      ✓ returns the last run status for a hook that has fired
      ✓ returns the last run status for triggerHook
      ✓ fails if no hook exists
    triggerHook
      ✓ should launch task with the given payload
      ✓ fails when creating the task fails
      ✓ fails if no hook exists
    resetTriggerToken
      ✓ creates a new token
      ✓ fails for undefined hook
    triggerHookWithToken
      ✓ successfully triggers task with the given payload
      ✓ should fail with invalid token
      ✓ fails with invalidated token
      ✓ fails with undefined hook
      ✓ trigger task after resetting the trigger token

  bin/schedule-hooks.js
    - schedule_hooks launches a scheduler

  Scheduler
    ✓ calls its poll method in a loop when started
    poll method
      ✓ calls handleHook only for past-due hooks
    handleHook method
firing hook tests/test with taskId KEPARfZBQo--jn1vxOlyRA
      ✓ creates a new task and updates nextTaskId, lastFire, nextScheduledDate
firing hook tests/test with taskId egoV1BkAQYuwe1aw0HSb4A
Failed to handle hook: tests/test, with err: Error: uhoh
      ✓ on error, sends an email and updates nextTaskId, lastFire, nextScheduledDate
firing hook tests/test with taskId Kx-4AD5_RXmL4Q38tCweTA
Failed to handle hook: tests/test, with err: Error: uhoh
      ✓ on error, notify is used with correct options

  TaskCreator
    ✓ firing a real task succeeds (1503ms)
    ✓ adds a taskId if one is not specified (1732ms)
    ✓ fails if task.scopes includes scopes not granted to the role (361ms)

  MockTaskCreator
    ✓ the fire method records calls

  validate
    ✓ create-hook-request.json
    ✓ schedule-none.json
    ✓ schedule-daily.json
    ✓ schedule-weekly.json
    ✓ schedule-weekday.json
    ✓ schedule-monthly.json
    ✓ schedule-biweekly.json


  68 passing (7s)
  1 pending
```